yt/frontends/gamer/fields.py

```diff
@@ -49,80 +50,207 @@ def __init__(self, ds, field_list):
 
     # add primitive and other derived variables
     def setup_fluid_fields(self):
+        pc = self.ds.units.physical_constants
         from yt.fields.magnetic_field import setup_magnetic_field_aliases
 
         unit_system = self.ds.unit_system
 
-        # velocity
-        def velocity_xyz(v):
-            def _velocity(field, data):
-                return data["gas", f"momentum_{v}"] / data["gas", "density"]
+        if self.ds.srhd:
```

Comment:
can this go into a separate function, like `_setup_srhd` or something?

Reply:
The only potential issue with that is that I'm trying to define different field functions whilst keeping only one `add_field` call for each one in this file.